[ISSUE] Issue with `databricks_connection` resource throwing `Error: cannot read connection: invalid ID:` after successfully creating a connection

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the Databricks Provider issue tracker reserved for bug reports and feature requests. For general usage questions, please see: <https://www.terraform.io/community.html>.
-->
When creating a new external connection for Lakehouse Federation, terraform creates the resource but fails to retrieve the ID, throwing the message:
```
│ Error: cannot read connection: invalid ID: 
│ 
│   with databricks_connection.postgresql,
│   on unity_catalog_management.tf line 8, in resource "databricks_connection" "postgresql":
│    8: resource "databricks_connection" "postgresql" {
│ 
╵
Releasing state lock. This may take a few moments...
```
### Configuration
<!-- Please provide a minimal reproducible configuration for the issue -->
```hcl
terraform {
  required_providers {

    databricks = {
      source  = "databricks/databricks"
      version = "1.35.0"
    }
}
}

provider "databricks" {
  auth_type = "azure-cli"
  alias     = "workspace"
  host      = module.databricks_workspace.workspace_url
}

resource "databricks_connection" "postgresql" {
  provider        = databricks.workspace
  name            = "lakehouse_federation_postgresql_tf"
  connection_type = "POSTGRESQL"
  comment         = "PostgreSQL Lakehouse Federation Connection"
  options = {
    host     = "HOST"
    port     = "5432"
    user     = user
    password = pwd
  }
}
```

### Expected Behavior
The connection should be created and imported into the terraform state file

### Actual Behavior
The connection was created successfully, but failed to be imported into the state file

### Steps to Reproduce
1. `terraform apply`-->

### Terraform and provider versions
Terraform version: 1.6.6
Provider version: 1.35.0

### Is it a regression?
Yes, it worked with the version 1.34 of the provider

### Debug Output
```
2024-01-26T10:54:15.241-0300 [INFO]  Starting apply for databricks_connection.postgresql
2024-01-26T10:54:15.241-0300 [DEBUG] databricks_connection.postgresql: applying the planned Create change
2024-01-26T10:54:16.996-0300 [DEBUG] provider.terraform-provider-databricks_v1.35.0: POST /api/2.1/unity-catalog/connections
> {
>   "comment": "PostgreSQL Lakehouse Federation Connection",
>   "connection_type": "POSTGRESQL",
>   "name": "lakehouse_federation_postgresql_tf",
>   "options": {
>     "host": "HOST.postgres.database.azure.com",
>     "password": "**REDACTED**",
>     "port": "5432",
>     "user": "databricks_lakehouse_federation"
>   }
> }
< HTTP/2.0 200 OK
< {
<   "comment": "PostgreSQL Lakehouse Federation Connection",
<   "connection_id": "CONNECTION_ID",
<   "connection_type": "POSTGRESQL",
<   "created_at": 1706277256985,
<   "created_by": "CREATOR_USERNAME",
<   "credential_type": "USERNAME_PASSWORD",
<   "full_name": "lakehouse_federation_postgresql_tf",
<   "metastore_id": "METASTORE_ID",
<   "name": "lakehouse_federation_postgresql_tf",
<   "options": {
<     "host": "HOST",
<     "port": "5432"
<   },
<   "owner": "CREATOR_USERNAME",
<   "provisioning_info": {
<     "state": "ACTIVE"
<   },
<   "read_only": true,
<   "securable_kind": "CONNECTION_POSTGRESQL",
<   "securable_type": "CONNECTION",
<   "updated_at": 1706277256985,
<   "updated_by": "CREATOR_USERNAME",
<   "url": "jdbc://HOST.postgres.database.azure.com:5432/"
< }: @module=databricks tf_provider_addr=registry.terraform.io/databricks/databricks tf_req_id=1f0bcd59-4582-38d1-7a7c-d79ed1de6f33 tf_resource_type=databricks_connection @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/logger/logger.go:33 tf_rpc=ApplyResourceChange timestamp=2024-01-26T10:54:16.996-0300
2024-01-26T10:54:16.997-0300 [ERROR] provider.terraform-provider-databricks_v1.35.0: Response contains error diagnostic: tf_req_id=1f0bcd59-4582-38d1-7a7c-d79ed1de6f33 @module=sdk.proto diagnostic_severity=ERROR tf_resource_type=databricks_connection tf_proto_version=5.4 tf_provider_addr=registry.terraform.io/databricks/databricks tf_rpc=ApplyResourceChange @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/diag/diagnostics.go:62 diagnostic_detail="" diagnostic_summary="cannot read connection: invalid ID: " timestamp=2024-01-26T10:54:16.996-0300
2024-01-26T10:54:16.997-0300 [DEBUG] State storage *remote.State declined to persist a state snapshot
2024-01-26T10:54:16.997-0300 [ERROR] vertex "databricks_connection.postgresql" error: cannot read connection: invalid ID:
```

--- a/catalog/resource_connection.go
+++ b/catalog/resource_connection.go
@@ -90,6 +90,10 @@ func ResourceConnection() *schema.Resource {
 			// We need to preserve original sensitive options as API doesn't return them
 			var cOrig catalog.CreateConnection
 			common.DataToStructPointer(d, s, &cOrig)
+			// If there are no options returned, need to initialize the map
+			if conn.Options == nil {
+				conn.Options = map[string]string{}
+			}
 			for key, element := range cOrig.Options {
 				if slices.Contains(sensitiveOptions, key) {
 					conn.Options[key] = element

--- a/catalog/resource_connection.go
+++ b/catalog/resource_connection.go
@@ -56,20 +56,19 @@ func ResourceConnection() *schema.Resource {
 			}
 			var createConnectionRequest catalog.CreateConnection
 			common.DataToStructPointer(d, s, &createConnectionRequest)
-			_, err = w.Connections.Create(ctx, createConnectionRequest)
+			conn, err := w.Connections.Create(ctx, createConnectionRequest)
 			if err != nil {
 				return err
 			}
 			// Update owner if it is provided
-			if d.Get("owner") == "" {
-				return nil
-			}
-			var updateConnectionRequest catalog.UpdateConnection
-			common.DataToStructPointer(d, s, &updateConnectionRequest)
-			updateConnectionRequest.NameArg = updateConnectionRequest.Name
-			conn, err := w.Connections.Update(ctx, updateConnectionRequest)
-			if err != nil {
-				return err
+			if d.Get("owner") != "" {
+				var updateConnectionRequest catalog.UpdateConnection
+				common.DataToStructPointer(d, s, &updateConnectionRequest)
+				updateConnectionRequest.NameArg = updateConnectionRequest.Name
+				conn, err = w.Connections.Update(ctx, updateConnectionRequest)
+				if err != nil {
+					return err
+				}
 			}
 			d.Set("metastore_id", conn.MetastoreId)
 			pi.Pack(d)

--- a/internal/acceptance/connection_test.go
+++ b/internal/acceptance/connection_test.go
@@ -22,20 +22,31 @@ func connectionTemplateWithOwner(host string, owner string) string {
 	`, host, owner)
 }
 
-func connectionTemplateWithoutOwner(host string) string {
-	return fmt.Sprintf(`
+func connectionTemplateWithoutOwner() string {
+	return `
 	resource "databricks_connection" "this" {
 		name = "name-{var.STICKY_RANDOM}"
-		connection_type = "MYSQL"
-		comment         = "this is a connection to mysql db"
-		options         = {
-			host     = "%s"
-			port     = "3306"
-			user     = "user"
-			password = "password"
+		connection_type = "BIGQUERY"
+		comment         = "test"
+		options = {
+			GoogleServiceAccountKeyJson = <<-EOT
+				{
+					"type": "service_account",
+					"project_id": "PROJECT_ID",
+					"private_key_id": "KEY_ID",
+					"private_key": "-----BEGIN PRIVATE KEY-----\nPRIVATE_KEY\n-----END PRIVATE KEY-----\n",
+					"client_email": "SERVICE_ACCOUNT_EMAIL",
+					"client_id": "CLIENT_ID",
+					"auth_uri": "https://accounts.google.com/o/oauth2/auth",
+					"token_uri": "https://accounts.google.com/o/oauth2/token",
+					"auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+					"client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/SERVICE_ACCOUNT_EMAIL",
+					"universe_domain": "googleapis.com"
+				}
+			EOT
 		}
 	}
-	`, host)
+	`
 }
 func TestUcAccConnectionsResourceFullLifecycle(t *testing.T) {
 	unityWorkspaceLevel(t, step{
@@ -49,8 +60,8 @@ func TestUcAccConnectionsResourceFullLifecycle(t *testing.T) {
 
 func TestUcAccConnectionsWithoutOwnerResourceFullLifecycle(t *testing.T) {
 	unityWorkspaceLevel(t, step{
-		Template: connectionTemplateWithoutOwner("test.mysql.database.azure.com"),
+		Template: connectionTemplateWithoutOwner(),
 	}, step{
-		Template: connectionTemplateWithoutOwner("test.mysql.database.aws.com"),
+		Template: connectionTemplateWithoutOwner(),
 	})
 }

--- a/internal/acceptance/connection_test.go
+++ b/internal/acceptance/connection_test.go
@@ -21,6 +21,22 @@ func connectionTemplateWithOwner(host string, owner string) string {
 	}
 	`, host, owner)
 }
+
+func connectionTemplateWithoutOwner(host string) string {
+	return fmt.Sprintf(`
+	resource "databricks_connection" "this" {
+		name = "name-{var.STICKY_RANDOM}"
+		connection_type = "MYSQL"
+		comment         = "this is a connection to mysql db"
+		options         = {
+			host     = "%s"
+			port     = "3306"
+			user     = "user"
+			password = "password"
+		}
+	}
+	`, host)
+}
 func TestUcAccConnectionsResourceFullLifecycle(t *testing.T) {
 	unityWorkspaceLevel(t, step{
 		Template: connectionTemplateWithOwner("test.mysql.database.azure.com", "account users"),
@@ -28,5 +44,13 @@ func TestUcAccConnectionsResourceFullLifecycle(t *testing.T) {
 		Template: connectionTemplateWithOwner("test.mysql.database.aws.com", "account users"),
 	}, step{
 		Template: connectionTemplateWithOwner("test.mysql.database.azure.com", "{env.TEST_METASTORE_ADMIN_GROUP_NAME}"),
+	})
+}
+
+func TestUcAccConnectionsWithoutOwnerResourceFullLifecycle(t *testing.T) {
+	unityWorkspaceLevel(t, step{
+		Template: connectionTemplateWithoutOwner("test.mysql.database.azure.com"),
+	}, step{
+		Template: connectionTemplateWithoutOwner("test.mysql.database.aws.com"),
 	})
 }

